### DISCUSSION TITLE
I cant find the implementation of SplitCHWRes in the source code

### DISCUSSION THREAD

**Discussion by snowbugxs** [idaholab/moose#29546] [Q&A Modules: Phase field] [answered]
For the split C-H equation, SplitCHWRes is designed to calculate the term of (M∇u,∇ψ). 
I find the implementation of this class is missing in the souce file, located at   ```/moose/modules/phase_field/src/kernels/SplitCHWRes.C```. 
Also, it's unusual the implementation of the base class SplitCHWResBase is not in the .C file but in the .h file.

**Comment by GiudGiud** [comment]
Hello

The implementation of the template is in the header because that's what is commonly done in C++ when you want to define a template that others may instantiate with types you did not foresee could be used in the template.

Guillaume

**Reply by snowbugxs** [reply]
Thanks. Sorry for I may cause a misunderstanding here.   
Both files ```SplitCHWRes.C``` and ```SplitCHWRes.h``` don't include the implementation of how to calculate residuals. 
Please check the two files.

**Reply by GiudGiud** [reply, marked as answer]
it is implemented in SplitCHWResBase.h

**Reply by snowbugxs** [reply]
Get it. Derived class inherits the methods defined in Base class.
